lib/liquid.rb

```diff
@@ -39,10 +39,16 @@ module Liquid
   SpacelessFilter             = /^(?:'[^']+'|"[^"]+"|[^'"])*#{FilterSeparator}(?:#{StrictQuotedFragment})(?:#{FirstFilterArgument}(?:#{OtherFilterArgument})*)?/o
   Expression                  = /(?:#{QuotedFragment}(?:#{SpacelessFilter})*)/o
   TagAttributes               = /(\w+)\s*\:\s*(#{QuotedFragment})/o
-  AnyStartingTag              = /\{\{|\{\%/
+  AnyStartingTag              = /#{TagStart}|#{VariableStart}/o
   PartialTemplateParser       = /#{TagStart}.*?#{TagEnd}|#{VariableStart}.*?#{VariableIncompleteEnd}/o
   TemplateParser              = /(#{PartialTemplateParser}|#{AnyStartingTag})/o
   VariableParser              = /\[[^\]]+\]|#{VariableSegment}+\??/o
+  WhiteSpaceControl           = /-/
+  WhiteSpaceTrimCheck         = /(#{TagStart}|#{VariableStart})#{WhiteSpaceControl}|#{WhiteSpaceControl}(#{TagEnd}|#{VariableEnd})/o
+  WhiteSpaceTagStart          = /[ \t]*\r?\n?[ \t]*(#{TagStart}|#{VariableStart})#{WhiteSpaceControl}/o
+  WhiteSpaceTagEnd            = /#{WhiteSpaceControl}(#{TagEnd}|#{VariableEnd})[ \t]*\r?\n?[ \t]*/o
```

Comment:
We don't need to adhere to erb. We should go with what's expected by
designers.
On Fri, Apr 15, 2016 at 1:27 PM Dylan Thacker-Smith <
notifications@github.com> wrote:

> In lib/liquid.rb
> <https://github.com/Shopify/liquid/pull/743#discussion_r59909832>:
>
> >    PartialTemplateParser       = /#{TagStart}.*?#{TagEnd}|#{VariableStart}.*?#{VariableIncompleteEnd}/o
> >    TemplateParser              = /(#{PartialTemplateParser}|#{AnyStartingTag})/o
> >    VariableParser              = /\[[^\]]+\]|#{VariableSegment}+\??/o
> > +  WhiteSpaceControl           = /-/
> > +  WhiteSpaceTrimCheck         = /(#{TagStart}|#{VariableStart})#{WhiteSpaceControl}|#{WhiteSpaceControl}(#{TagEnd}|#{VariableEnd})/o
> > +  WhiteSpaceTagStart          = /[ \t]*\r?\n?[ \t]*(#{TagStart}|#{VariableStart})#{WhiteSpaceControl}/o
> > +  WhiteSpaceTagEnd            = /#{WhiteSpaceControl}(#{TagEnd}|#{VariableEnd})[ \t]*\r?\n?[ \t]*/o
>
> This isn't really ERB like in its behaviour. In ERB it trims a following
> newline, but doesn't trim spaces after it. Similarly, the trim on the start
> tag trims up to the newline, but not including it in ERB.
>
> This is also different from go templates, where it will trim all preceding
> or following whitespace.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/liquid/pull/743/files/a01ac62c891a653692a950e804fc453ded7ebd78#r59909832>
>
-- 
- tobi
CEO Shopify